### PR TITLE
fixing get user_id issue

### LINKED ISSUE #702: BUG: GET/users/{user_id} returns list of null users for id that doesn't exist
**Describe the bug**
When a user trying to view other user's details by sending GET /users/{user_id} with user_id that does not exist, the system returns a list of null users instead of HTTP error 404 (Not Found)

**To Reproduce**

1. Login as a user
2. Go to GET users/{user_id} and enter a random id that does not exist in the Users table (check the database to do this)
3. See the response with list of empty users with null ids

**Expected behavior**
Must return http eerror 404 Not Found and message that user does not exist

**Screenshots**
Screen Shot 2020-07-29 at 5 10 27 pm

**Desktop (please complete the following information):**
 - OS: MacOS
 - Browser Safari
 - Version ?

### PR DESCRIPTION
**Pull request by saikirankv**
### Description

Fixes #702 

- Code

**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)

### How Has This Been Tested?

yes, I have tested different scenarios in local to make sure everything was working fine.
1. passing a valid user id
2. passing invalid user id

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials

**Code/Quality Assurance Only**

- [x] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been published in downstream modules

### CODE REVIEW

**Comment by saikirankv** [comment]
@isabelcosta @vj-codes can you review this PR?

**Comment by saikirankv** [comment]
@SanketDG

File: app/api/resources/user.py [hunk 1]
```diff
@@ -90,12 +90,11 @@ def get(cls, user_id):
         A user with valid access token can view the details of another user. The endpoint
         takes "user_id" of such user has input.
         """
-        requested_user = DAO.get_user(user_id)
-        if requested_user is None:
-            return messages.USER_DOES_NOT_EXIST, HTTPStatus.NOT_FOUND
+        user_details = DAO.get_user(user_id)
```
Comment (devkapilbansal) [inline app/api/resources/user.py]: Change the name back to requested_user

**Review by devkapilbansal — CHANGES_REQUESTED** [review]
Thanks @saikirankv for opening this.
But I can see that you change the variable name from `requested_user` to `user_details`. Is it intentional??
If not please revert it back.

Also, squash your commits into one

**Comment by saikirankv** [comment]
@devkapilbansal I made the necessary changes, can you review again. Sorry I have deleted branch in my local, can please do squash and merge of this PR?

**Comment by devkapilbansal** [comment]
> @devkapilbansal I made the necessary changes, can you review again. Sorry I have deleted branch in my local, can please do squash and merge of this PR?

You can fetch that branch again in local. I can't do the changes

**Comment by isabelcosta** [comment]
@saikirankv can you please add information on PR description on how you tested this feature please? It can be helpful for a QA contributor to test your fix.
@NenadPantelic can you help reviewing this PR? (asking this because of your comment https://github.com/anitab-org/mentorship-backend/issues/702#issuecomment-761806453)
@devkapilbansal I can use "Squash and Merge" option in the end when this gets approved and tested/

**Comment by NenadPantelic** [comment]
Tested locally:
1) unit tests pass (all of them)
2) tested through Swagger UI:
- Case 1: existing user (id = 2)
image
- Case 2: non-existing user (id=100)
image
Code style followed properly.

**Review by NenadPantelic — COMMENTED** [review]
The code is good. Please update your PR description and fill in the proper options.

**Comment by saikirankv** [comment]
updated the PR description.

**Comment by NenadPantelic** [comment]
> updated the PR description.

Hi @saikirankv
1) update the description
2) remove the things you don't need -> bugfix or new feature but not both
3) add screenshots from the Swagger with both scenarios - valid and invalid id etc.

**Comment by saikirankv** [comment]
Updated the PR description, I think you already added the screenshots.

**Review by NenadPantelic — APPROVED** [review]
LGTM :+1:

**Comment by NenadPantelic** [comment]
@vj-codes @devkapilbansal @Aaishpra @codesankalp Can someone review it?

**Review by Aaishpra — APPROVED** [review]
Great work @saikirankv 🎉 and thanks for the testing @NenadPantelic

**Comment by codesankalp** [comment]
The changes made in this PR were tested locally. Following are the results:

1. Code review - Done.

2. All possible responses (positive and negative tests) were tested as below:

  * _Test1 Description_  : Checking for user id that does not exist.
    _Screenshot/gif_:  
      User Table: 
     image
      API Result: 
     oie_vJkAvZQvIR4Q

    _Expected Result_:  404 Not Found.
    _Actual Result_: 404 Not Found.
    
3. Additional Comments:
Tested and reviewed. LGTM :tada: 

@isabelcosta @saikirankv

**Review by isabelcosta — APPROVED** [review]
Thank you for your contribution @saikirankv !

**Pull request merged**